### PR TITLE
Add section on overflow

### PR DESCRIPTION
**Pull request by chiphogg**
This largely adapts the source material from the [Au docs](https://aurora-opensource.github.io/au/0.4.1/discussion/concepts/overflow/), fixing it up for the standard units proposal context, and adding a subsection on using overflow-safe reps.

I decided to make concrete recommendations: of the 5 approaches considered, 2 are unfeasible or inappropriate, and the other 3 are complementary.

I also fixed up a couple of markdown errors that were messing with the syntax highlighting in my editor; these are done in a separate commit.

### CODE REVIEW

File: src/3045R5_quantities_and_units_library.md [hunk 1]
```diff
@@ -3991,9 +3991,9 @@ of them together with the recommended portable replacements:
 | GREEK SMALL LETTER PI |   π    | u8"\u03c0" |         "pi"         |
 | DOT OPERATOR          |   ⋅    | u8"\u22C5" |      `<none>`*       |
 
-(*) Users should not select `unit_symbol_separator::half_high_dot` and `character_set::portable`
+_NOTE: Users should not select `unit_symbol_separator::half_high_dot` and `character_set::portable`
```
Comment (mpusz) [inline src/3045R5_quantities_and_units_library.md:3994]: This star referred to line 3992, which had this star as well. Do you have a better idea of how to connect those?

File: src/3045R5_quantities_and_units_library.md [hunk 2]
```diff
@@ -4467,7 +4467,7 @@ library does not have enough information to print it that way by itself.
 5. `std::chrono::duration` uses 'Q' and 'q' for a number and a unit. In the grammar above, we
    proposed using 'N' and 'U' for them, respectively. We also introduced 'D' for dimensions. Are
    we OK with this?
-6. Are we OK with the usage of '_' for denoting a subscript identifier? Should we use it everywhere
+6. Are we OK with the usage of `_` for denoting a subscript identifier? Should we use it everywhere
```
Comment (mpusz) [inline src/3045R5_quantities_and_units_library.md:4470]: With single quotes, I wanted to express that I mean a character, as was the case for `Q` and `q` in the previous point. Maybe we should use `'_'` instead everywhere?

File: src/3045R5_quantities_and_units_library.md [hunk 3]
```diff
@@ -5165,6 +5164,172 @@ Having different operators for safe floating-point operations and unsafe integer
 hurt generic programming.
 As such, users should instead use safer representation types.
 
+### Integer overflow
+
+To convert a quantity in a program to different units, we need to multiply or divide by a conversion
+factor.  Sometimes, the result is too big to fit in the type: a problem known as _overflow_.  While
```
Comment (mpusz) [inline src/3045R5_quantities_and_units_library.md:5170]:
```suggestion
To convert a quantity to different units, we must multiply or divide its numerical value by a conversion
factor.  Sometimes, the result is too big to fit in the type: a problem known as _overflow_.  While
```

File: src/3045R5_quantities_and_units_library.md [hunk 4]
```diff
@@ -5165,6 +5164,172 @@ Having different operators for safe floating-point operations and unsafe integer
...
+any numerical type can overflow, we focus on integral types here, because they carry the most risk.
+Even the smallest floating point type in common use, `float`, has a range of $10^{38}$, while the
+diameter of the observable universe measured in atomic diameters is "only" about $10^{37}$![^333]
+
+[^333]: Here, we take the radius of the observable universe as 46.6 billion light years, and the
```
Comment (mpusz) [inline src/3045R5_quantities_and_units_library.md:5175]: Why do you use `^333` here and `^2` below? Did you mean `^1` here?

**Review by mpusz — COMMENTED** [review]
Thanks!

**Review by JohelEGP — APPROVED** [review]
This is great.

File: src/3045R5_quantities_and_units_library.md (hunk 1 above)
Comment (chiphogg) [reply src/3045R5_quantities_and_units_library.md]: I didn't know what the star was there for, but this makes sense.  I switched to the same footnote notation I was using elsewhere.  I verified that it works in the rendered output.

File: src/3045R5_quantities_and_units_library.md (hunk 2 above)
Comment (chiphogg) [reply src/3045R5_quantities_and_units_library.md]: Done.

File: src/3045R5_quantities_and_units_library.md (hunk 4 above)
Comment (chiphogg) [reply src/3045R5_quantities_and_units_library.md]: Hah... that was just me being lazy.  :sweat_smile:  I didn't know if we had any footnotes before, but I was pretty sure we wouldn't have 333 of them!  I cleaned up the footnote numbering in the latest commits.

File: src/3045R5_quantities_and_units_library.md [hunk 5]
```diff
@@ -3989,11 +3989,11 @@ of them together with the recommended portable replacements:
...
 | MULTIPLICATION SIGN   |   ×    | u8"\u00d7" |         "x"          |
 | GREEK SMALL LETTER PI |   π    | u8"\u03c0" |         "pi"         |
-| DOT OPERATOR          |   ⋅    | u8"\u22C5" |      `<none>`*       |
+| DOT OPERATOR          |   ⋅    | u8"\u22C5" |      `<none>`[^1]    |
 
-(*) Users should not select `unit_symbol_separator::half_high_dot` and `character_set::portable`
+[^1]: Users should not select `unit_symbol_separator::half_high_dot` and `character_set::portable`
 at the same time. This symbol is valid only for UTF-8 encoding. Otherwise, we propose to throw
-an exception during the unit symbol string processing.
+an exception during the unit symbol string processing._
```
Comment (mpusz) [inline src/3045R5_quantities_and_units_library.md:3996]:
```suggestion
an exception during the unit symbol string processing.
```

**Review by mpusz — CHANGES_REQUESTED** [review]

**Review by mpusz — APPROVED** [review]

**Pull request merged**